tests/goth_tests/test_mid_agreement_payments/test_mid_agreement_payments.py

```diff
@@ -0,0 +1,50 @@
+"""A goth test scenario for mid-agreement payments."""
+from functools import partial
+import logging
+import os
+from pathlib import Path
+import re
+from typing import List
+
+import pytest
+
+from goth.configuration import load_yaml, Override
+from goth.runner.log import configure_logging
+from goth.runner import Runner
+from goth.runner.probe import RequestorProbe
+
+
+logger = logging.getLogger("goth.test.mid_agreement_payments")
+
+
+@pytest.mark.asyncio
+async def test_long_expiration(
+    log_dir: Path,
+    goth_config_path: Path,
+    config_overrides: List[Override],
+) -> None:
+
+    # This is the default configuration with 2 wasm/VM providers
+    goth_config = load_yaml(goth_config_path, config_overrides)
+    test_script_path = str(Path(__file__).parent / "requestor.py")
```

Comment:
we could also use an existing example - e.g. the simple service (or http_proxy) with `--running-time 37000` and just issue a SIGINT once we confirm the service has been started ...